### PR TITLE
Add dynamic thresholding to DDIM and DDPM

### PR DESCRIPTION
**Pull request by PeterL1n**
Only changed pytorch version. Not changed flax.

### CODE REVIEW

**Comment by PeterL1n** [comment]
https://github.com/huggingface/diffusers/issues/1636

**Comment by patrickvonplaten** [comment]
That's a super nice PR @PeterL1n, many people have asked for thresholding :-) 

I've added some comments. 
Also could we add two tests:
- 1. A test to all schedulers for which dynamic thresholding is added. The test can be as simple as: https://github.com/huggingface/diffusers/blob/1e216be8950d22813b975a4d026e58011cd78162/tests/test_scheduler.py#L640 just for different thresholding values
- 2. One slow test for Stable Diffusion that shows that thresholding leads to good image resutls (happy to help with this one)

File: src/diffusers/schedulers/scheduling_ddim.py [hunk 1]
```diff
@@ -295,9 +307,19 @@ def step(
...
 
-        # 4. Clip "predicted x_0"
+        # 4. Clip or threshold "predicted x_0"
         if self.config.clip_sample:
-            pred_original_sample = torch.clamp(pred_original_sample, -1, 1)
+            pred_original_sample = torch.clamp(pred_original_sample, -self.config.sample_max_value, self.config.sample_max_value)
+        
+        if self.config.thresholding:
+            # Dynamic thresholding in https://arxiv.org/abs/2205.11487
+            dynamic_max_val = pred_original_sample \
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddim.py:316]:
```suggestion
            pred_original_sample = self.apply_threshold(pred_orginal_sample)
```

Can we move this into it's own function? 

Then we can use the `# Copy` mechanism in all schedulers to copy the logic, e.g. as done here: https://github.com/huggingface/diffusers/blob/1b0b54a4264453818d7a6474d336a19922e3490a/src/diffusers/schedulers/scheduling_ddim.py#L31

This way everything stays in sync **and** is very readable :-)

Comment (patrickvonplaten) [reply src/diffusers/schedulers/scheduling_ddim.py:316]: Note that the function should be part of the Scheduler so that it has access to `self.dynamic_thresholding_ratio`

**Comment by patrickvonplaten** [comment]
Also cc @patil-suraj @pcuenca

**Comment by clarencechen** [comment]
@patil-suraj @patrickvonplaten I'll integrate your suggestions into this PR, write the tests, and carry it to the finish line. @patrickvonplaten Are there any tests currently computing FID or other perceptual metrics in the CI pipeline that you could point me to?

**Comment by patrickvonplaten** [comment]
Closing in favor of: https://github.com/huggingface/diffusers/pull/2528
